src/System.ComponentModel.TypeConverter/src/System/ComponentModel/BindingList.cs

```diff
@@ -0,0 +1,753 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+
+
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix", Scope = "type", Target = "System.ComponentModel.BindingList`1")]
+
+namespace System.ComponentModel
+{
+    using System;
+    using System.Reflection;
+    using System.Collections;
+    using System.Collections.ObjectModel;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Diagnostics;
+    using System.Security.Permissions;
+    using CodeAccessPermission = System.Security.CodeAccessPermission;
+
+    /// <include file='doc\BindingList.uex' path='docs/doc[@for="BindingList"]/*' />
+    /// <summary>
+    /// </summary>
+    [HostProtection(SharedState = true)]
+    [Serializable]
+    public class BindingList<T> : Collection<T>, IBindingList, ICancelAddNew, IRaiseItemChangedEvents
+    {
+        private int _addNewPos = -1;
+        private bool _raiseListChangedEvents = true;
+        private bool _raiseItemChangedEvents = false;
+
+        [NonSerialized()]
```

Comment:
Will leave as-is for now. We can clean up all the code (including the other suggestions) once the port is complete!